.gitignore

```diff
@@ -1,3 +1,5 @@
 build/*
 leftovers/*
 **/.DS_Store
+.vs/*
+out/*
```

Comment:
When running cmake through Visual Studio it defaults to out/build for the build directory and out/install for the install directory, I didn't feel it was worth overwriting these defaults, figured adding an additional ignore directory was reasonable.